allennlp/training/trainer.py

```diff
@@ -998,35 +535,28 @@ def _restore_checkpoint(self) -> Tuple[int, List[float]]:
             The epoch at which to resume training, which should be one after the epoch
             in the saved training state.
         """
-        latest_checkpoint = self.find_latest_checkpoint()
+        model_state, training_state = self._checkpointer.restore_checkpoint()
 
-        if latest_checkpoint is None:
+        if not training_state:
             # No checkpoint to restore, start at 0
-            return 0, []
-
-        model_path, training_state_path = latest_checkpoint
+            return 0
 
-        # Load the parameters onto CPU, then transfer to GPU.
-        # This avoids potential OOM on GPU for large models that
-        # load parameters onto GPU then make a new GPU copy into the parameter
-        # buffer. The GPU transfer happens implicitly in load_state_dict.
-        model_state = torch.load(model_path, map_location=util.device_mapping(-1))
-        training_state = torch.load(training_state_path, map_location=util.device_mapping(-1))
         self.model.load_state_dict(model_state)
         self.optimizer.load_state_dict(training_state["optimizer"])
         if self._learning_rate_scheduler is not None and "learning_rate_scheduler" in training_state:
-            self._learning_rate_scheduler.lr_scheduler.load_state_dict(
-                    training_state["learning_rate_scheduler"])
-        move_optimizer_to_cuda(self.optimizer)
-
-        # We didn't used to save `validation_metric_per_epoch`, so we can't assume
-        # that it's part of the trainer state. If it's not there, an empty list is all
-        # we can do.
-        if "val_metric_per_epoch" not in training_state:
-            logger.warning("trainer state `val_metric_per_epoch` not found, using empty list")
-            val_metric_per_epoch: List[float] = []
+            self._learning_rate_scheduler.lr_scheduler.load_state_dict(training_state["learning_rate_scheduler"])
+        training_util.move_optimizer_to_cuda(self.optimizer)
+
+        # Currently the ``training_state`` contains a serialized ``MetricTracker``.
+        if "metric_tracker" in training_state:
+            self._metric_tracker.load_state_dict(training_state["metric_tracker"])
+        # It used to be the case that we tracked ``val_metrics_per_epoch``.
```

Comment:
@joelgrus, it seems there is a small typo here. It should be `val_metric_per_epoch`. Fixed this, and added a test for it in #2352.